components/sst_importer/src/import_mode2.rs

```diff
@@ -0,0 +1,302 @@
+            inner,
+            is_regional_import,
+        }
+    }
+
+    // Periodically perform timeout check to change import mode of some regions back
+    // to normal mode.
+    pub fn start(&self, executor: &Handle) {
+        // spawn a background future to put TiKV back into normal mode after timeout
+        let inner = self.inner.clone();
+        let switcher = Arc::downgrade(&inner);
+        let timer_loop = async move {
+            let mut prev_range = None;
+            // loop until the switcher has been dropped
+            while let Some(switcher) = switcher.upgrade() {
+                let next_check = {
+                    let now = Instant::now();
+                    let mut switcher = switcher.lock().unwrap();
+                    if let Some(range) = prev_range.take() {
+                        if let Some(next_check) = switcher.next_check.get(&range) {
+                            if now >= *next_check {
+                                switcher.clear_import_mode_regions_in_range(range);
+                            }
+                        }
+                    }
+
+                    let mut min_next_check = now + switcher.timeout;
+                    for (range, next_check) in &switcher.next_check {
+                        if *next_check < min_next_check {
+                            min_next_check = *next_check;
+                            prev_range = Some(range.clone());
+                        }
+                    }
+                    min_next_check
+                };
+
+                let ok = GLOBAL_TIMER_HANDLE.delay(next_check).compat().await.is_ok();
+                if !ok {
+                    warn!("failed to delay with global timer");
+                }
+            }
+        };
+        executor.spawn(timer_loop);
+    }
+
+    pub fn enter_import_mode_for_regions_in_range(
+        &self,
+        range: KeyRange,
+        store_regions_info: &HashMap<u64, (Region, bool)>,
+    ) {
+        let range = Range::from(range);
+        let mut inner = self.inner.lock().unwrap();
+        let mut regions = HashSet::default();
+        for (region_id, (region, _)) in store_regions_info {
+            if region_overlap_with_range(&range, region) {
+                regions.insert(*region_id);
+            }
+        }
+        if regions.is_empty() {
+            return;
+        }
+        inner.is_regional_import.store(true, Ordering::Release);
+        let next_check = Instant::now() + inner.timeout;
+        // range may exist before, but store_regions_info may have been changed
+        inner.import_mode_regions.insert(range.clone(), regions);
+        inner.next_check.insert(range, next_check);
+    }
+
+    pub fn clear_import_mode_regions_in_range(&self, range: KeyRange) {
+        let mut inner = self.inner.lock().unwrap();
+        let range = Range::from(range);
+        inner.clear_import_mode_regions_in_range(range);
+    }
+
+    pub fn regional_import_mode(&self) -> bool {
```

Comment:
I think we should have region_id as the parameter to precisely know if the region is in import mode or not.